functions/captureChannelWithBot.protected.ts

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright (C) 2021-2023 Technology Matters
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+/* eslint-disable global-require */
+/* eslint-disable import/no-dynamic-require */
+import '@twilio-labs/serverless-runtime-types';
+import { Context, ServerlessCallback } from '@twilio-labs/serverless-runtime-types/types';
+import {
+  responseWithCors,
+  bindResolve,
+  error400,
+  error500,
+  success,
+} from '@tech-matters/serverless-helpers';
+import AWS from 'aws-sdk';
+import type { MessageInstance } from 'twilio/lib/rest/chat/v2/service/channel/message';
+
+type EnvVars = {
+  CHAT_SERVICE_SID: string;
+  ASELO_APP_ACCESS_KEY: string;
```

Comment:
I've meant to ask this question before in other code, but does twilio serverless set its own AWS creds? Is that why we use non-standard ENV vars?

If we could just use `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and `AWS_REGION` there would be [no reason to do any manual configuration](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/loading-node-credentials-environment.html) of the client.

Reply:
Oh. These only come in via context, not as .env files, huh? Or do they? The [wording in the docs is confusing](https://www.twilio.com/docs/labs/serverless-toolkit/developing#environment-variables).

Reply:
I think `ASELO_APP_ACCESS_KEY` is just the name we gave to this AWS credentials, to emphasize the fact that they are not the credentials `AWS_ACCESS_KEY_ID` used by Github Actions, but a different that correspond to the "aselo app user" with very limited capabilities. 

Is that your questions? Sorry if I misunderstood.

Reply:
There is one real question:

Are these "env" vars that are being loaded from `context` also available at `process.env. ASELO_APP_ACCESS_KEY `?

And one statement: 

If they *are* also available inside of the serverless function in the `process.env` object, and they are renamed in serverless to the ENV vars expected by aws-sdk (`AWS_ACCESS_KEY_ID`, and `AWS_SECRET_ACCESS_KEY`) then there is no reason to have any of the blocks like this:

```
AWS.config.update({
      credentials: {
        accessKeyId: context.ASELO_APP_ACCESS_KEY,
        secretAccessKey: context.ASELO_APP_SECRET_KEY,
      },
      region: context.AWS_REGION,
    });
```

aws-sdk will automatically read the env vars if they are set to the expected name.